builtin/logical/pki/crl_util.go

```diff
@@ -219,11 +302,60 @@ func buildCRL(ctx context.Context, b *backend, req *logical.Request, forceNew bo
 		} else {
 			newRevCert.RevocationTime = time.Unix(revInfo.RevocationTime, 0).UTC()
 		}
-		revokedCerts = append(revokedCerts, newRevCert)
+
+		// Now we need to assign the revoked certificate to an issuer.
+		var foundParent = false
+		for issuerId, issuerCert := range issuerIdCertMap {
+			if bytes.Equal(revokedCert.RawIssuer, issuerCert.RawSubject) {
+				if err := revokedCert.CheckSignatureFrom(issuerCert); err == nil {
+					// Valid mapping. Add it to the specified entry.
+					revokedCertsMap[issuerId] = append(revokedCertsMap[issuerId], newRevCert)
+					foundParent = true
+					break
+				}
+			}
+		}
+
+		if !foundParent {
+			unassignedCerts = append(unassignedCerts, newRevCert)
+		}
+	}
+
+	return unassignedCerts, revokedCertsMap, nil
+}
+
+// Builds a CRL by going through the list of revoked certificates and building
+// a new CRL with the stored revocation times and serial numbers.
+func buildCRL(ctx context.Context, b *backend, req *logical.Request, forceNew bool, thisIssuerId issuerId, revoked []pkix.RevokedCertificate) error {
+	crlInfo, err := b.CRL(ctx, req.Storage)
+	if err != nil {
+		return errutil.InternalError{Err: fmt.Sprintf("error fetching CRL config information: %s", err)}
+	}
+
+	crlLifetime := b.crlLifetime
+	var revokedCerts []pkix.RevokedCertificate
```

Comment:
Not sure why we are defining another variable here, we simply always pass through the `revoked` param? Am I missing something?

Reply:
This is expected; see the complex conditionals at the start of `buildCRL` -- there's several cases where we write out a signed (but empty-list) CRL, even if we may have revoked certs.